src/components/Login/LoginFormWalletFile/LoginFormWalletFile.js

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import { remote } from 'electron';
+import { bool, string, func } from 'prop-types'; import { noop } from 'lodash';
+import { wallet } from '@cityofzion/neon-js';
+
+import Button from '../../Forms/Button';
+import styles from './LoginFormWalletFile.scss';
+
+
+
+export default class LoginFormWalletFile extends React.Component {
+
+  static propTypes = {
+    disabled: bool,
+    wif: string,
+    onLogin: func,
+  };
+
+  static defaultProps = {
+    disabled: false,
+    wif: '',
+    onLogin: noop,
+  };
+
+  state = {
+    wif: '',
+    accountsInWalletFile: []
+  }
+
+
+
+
+
+  handleLoadWallet = () => {
+    var showOpenDialogOptions = {
+        title: `Select Wallet file`,
+        filters: [{ name: `Wallet file`, extensions: [`json`] }]
+      };
+    remote.dialog.showOpenDialog(showOpenDialogOptions, (selectedFile) => {
+      if (selectedFile) {
+        this.load(selectedFile[0]); //There can only be 1 value in the array as only 1 file can be loaded at a time
+      }
+    });
+  }
+
+
+  load = (selectedFile) => {
+    const { accountsInWalletFile } = this.state;
+    // const { walletFile } = this.props
+
+    try {
+      this.walletFile = wallet.Wallet.readFile(selectedFile);
+    } catch (err) {
+      this.props.alert(`Error reading the wallet file: ${err.message}`);
+      return
+    }
+
+    if (!this.walletFile.accounts.length > 0)
+    {
+      this.props.alert(`This wallet file has no account in it.`);
+      return
+    }
+
+    this.removeEverythingFromArray(accountsInWalletFile);
+
+    this.walletFile.accounts.forEach(function(account, i) {

```

Comment:
I'd use something like this,  `this.setState({ accountsInWalletFile: Object.assign({}, this.walletFile.accounts) })` copying the whole array to the state instead of looping over it and mutating the state

Reply:
Prefer ES6 syntax for that sample code though:

```js
this.setState({ accountsInWalletFile: { ...this.walletFile.accounts } })
```